CONTRIBUTING.md

```diff
@@ -1,6 +1,95 @@
+
+Hopefully there are no clashes with the proposed contribution and ongoing work and the proposed work
+can proceed.
+In the issue proposing the contribution click the `Create a branch` link, this will open a popup
+where you can name your branch and select your forked repository as the branch destination.
+At this point you will be able to checkout your branch locally and begin working.
+
+### Before you start
+
+If you have already worked on CCS then you should ensure your local repository view of `develop` is
+up to date before beginning new work.
+To do so we recommend using `git rebase` to keep the git history clean, assuming that the main
+ASiMoV-CCS repository is `origin` and your branch is called `my-branch`:
+```
+git checkout develop
+git pull origin/develop
+git checkout my-branch
+git rebase develop
+```
+
+You are now ready to begin.
+
+### Keeping up to date
+
+We recommend periodically pulling the `origin/develop` branch and rebasing your working branch onto
+`develop`.
+If the development of your contribution continues for some time, this will ensure that it does not
+stray too far from the eventual merge target.
+
+## Submitting a contribution
+
+Once you are satisfied with your contribution, open a pull request against `upstreaming` on the main
+repository, linking it to the issue where you proposed the change - if you followed the above steps
+to create a new branch this linking should occur automatically, except you may need to change the
+target branch of the pull request on GitHub.
+This will begin a process of further interaction with the ASiMoV-CCS developers, who may request
+changes before the contribution is accepted.
+
+Note that you don't need to have a "complete" contribution to open a pull request.
+Opening a draft pull request early in the development process to begin discussions on code changes
+can also be beneficial.
+
+# Instructions for core developers accepting contributions
+
+The process for accepting a contribution into the upstream GitLab repository is outlined below.
+Currently, a manual process is required to push the contribution to the upstream repository.
+
+0. When a potential contributor opens an issue regarding making a contribution:
+- is the proposed contribution well-described? Ask for clarification if not. **TODO:** develop a
+  template for bug reports and proposed changes.
+- check whether the proposed contribution conflicts or overlaps with any ongoing or planned work -
+  try to discuss this with the contributor, perhaps they could collaborate on any preexisting work?
+- if the proposed contribution looks good to proceed ensure that `upstreaming` is up to date with
+  `develop` by rebasing (see above instructions) and push the updated `upstreaming` branch to GitHub
+1. When a pull request is opened check it is targetting the `upstreaming` branch, change the target
+   branch if necessary.
+2. Ensure that the `upstreaming` branch is up to date by rebasing and pushing to GitHub if
```

Comment:
This is a duplicate of line 81

Reply:
Removed line 81 to clarify (original instructions had them work against upstreaming, now it is only important when they come to merge into upstreaming)